tools/src/verdaccio/registry.ts

```
@@ -89,92 +91,108 @@ export function nxStartVerdaccioServer({
+                },
+              };
+
+              console.info(
+                `${gray('>')} ${gray(
+                  bold('Verdaccio'),
+                )} Registry started on URL: ${bold(
+                  result.registry.url,
+                )}, with PID: ${bold(
+                  listProcess({ commandMatch: commandId }).at(0)?.pid,
+                )}`,
+              );
+              if (verbose) {
+                console.info(`${gray('>')} ${gray(bold('Verdaccio'))}`);
+                console.table(result);
+              }
+
+              resolve(result);
+            }
+          },
+          onStderr: (stderr: string) => {
+            if (verbose) {
+              process.stdout.write(
+                `${red('>')} ${red(bold('Verdaccio'))} ${stderr}`,
+              );
+            }
+          },
         },
-        onStderr: (data: string) => {
-          if (verbose) {
-            process.stdout.write(data);
+      })
+        // @TODO reconsider this error handling
+        .catch(error => {
+          if (error.message !== 'Failed to start verdaccio: undefined') {
+            console.error(
+              `${red('>')} ${red(
+                bold('Verdaccio'),
+              )} Error starting ${projectName} verdaccio registry:\n${error}`,
+            );
+          } else {
+            reject(error);
           }
-        },
-      },
+        });
     })
-      // @TOSO reconsider this error handling
-      .catch(error => {
-        if (error.message !== 'Failed to start verdaccio: undefined') {
-          console.error(
-            `Error starting ${projectName} verdaccio registry:\n${
-              error as Error
-            }`,
-          );
-          reject(error);
-        }
-        teardownTestFolder(storage);
-        throw error;
-      });
-  });
+      // in case the server dies unexpectedly clean folder
+      .catch(() => teardownTestFolder(storage))
```

Careful not to swallow the error.

```suggestion
      .catch((error: unknown) => {
        teardownTestFolder(storage);
        throw error;
      })
```